lib/icingadb/icingadb.cpp

```diff
@@ -203,6 +199,26 @@ void IcingaDB::Stop(bool runtimeRemoved)
 	ObjectImpl<IcingaDB>::Stop(runtimeRemoved);
 }
 
+void IcingaDB::ValidateUsername(const Lazy<String>& lvalue, const ValidationUtils& utils)
+{
+	ObjectImpl<IcingaDB>::ValidateUsername(lvalue, utils);
+
+	if (!lvalue().IsEmpty() && GetPassword().IsEmpty()) {
+		BOOST_THROW_EXCEPTION(ValidationError(this, { "username" },
+			"Redis password must be set, if username is provided."));
```

Comment:
Here there are only two. But still I could update (w/o no_user_modify) any of them causing this check to fail:

* Set username while there's no password
* Unset password while there's a username

If you insist, put this in ValidatePassword() at least IMAO.

Reply:
> But still I could update (w/o no_user_modify) any of them causing this check to fail:

I'm having a hard time understanding what you're trying to imply as to what the problem is, first of all, what does `w/o no_user_modify` mean? Did you remove the `no_user_modify` restriction? The `password` attribute is never supposed to be changed at runtime, nor is `username`. So, if you remove this restriction, what do you expect to happen then?